lib/src/dartdoc_options.dart

```diff
@@ -1120,19 +1121,18 @@ abstract class _DartdocFileOption<T> implements DartdocOption<T> {
   _OptionValueWithContext<Object> _valueAtFromFile(Folder dir) {
     var yamlFileData = _yamlAtDirectory(dir);
     var contextPath = yamlFileData.canonicalDirectoryPath;
-    dynamic yamlData = yamlFileData.data ?? {};
+    Object yamlData = yamlFileData.data ?? {};
     for (var key in keys) {
-      if (!yamlData.containsKey(key)) return null;
-      yamlData = yamlData[key] ?? {};
+      if (yamlData is Map && !yamlData.containsKey(key)) return null;
+      yamlData = (yamlData as Map)[key] ?? {};
     }
 
-    dynamic returnData;
+    Object returnData;
     if (_isListString) {
       if (yamlData is YamlList) {
-        returnData = <String>[];
-        for (var item in yamlData) {
-          returnData.add(item.toString());
-        }
+        returnData = [
```

Comment:
I think this is fine, though it seems odd to me to omit `<String>`?

Reply:
`String` is inferred from the elements as the type of the list. This is listed in  Effective Dart:

https://dart.dev/guides/language/effective-dart/design#dont-write-type-arguments-on-generic-invocations-that-are-inferred